src/blocks/sandbox-image/edit.js

```diff
@@ -0,0 +1,260 @@
+		},
+	];
+
+	// replaces wp image? https://github.com/WordPress/gutenberg/blob/c54fa0beb059a2e3b2d2f5a32f26ab47598be0b6/packages/block-library/src/image/edit.js
+	return (
+		<>
+			<div { ...useBlockProps() }>
+				<Image
+					// Component setup
+					canEditImage={ attributes.canEditImage }
+					debug={ true }
+					tag={ attributes.tag }
+					size={ attributes.size }
+					sourceSizes={ demo_sourceSizes }
+					altSource={ attributes.altSource }
+					mediaId={ demo_mediaId }
+					className={ attributes.className }
+					canEditFocalPoint={ attributes.canEditFocalPoint }
+					focalPoint={ demo_focalPoint }
+					//
+					// Placeholder overrrides
+					label={ attributes.label }
+					instructions={ attributes.instructions }
+					//
+					// MediaPlaceholder overrrides
+					labelsMediaPlaceholder={ demo_labelsMediaPlaceholder }
+				/>
+			</div>
+
+			{
+				<InspectorControls>
+					<PanelBody title={ __( 'Block Configurator' ) }>
+						<PanelRow>
+							Use these controls to adjust the parameters sent to
+							the Component:
+						</PanelRow>
+						<PanelRow>Component setup</PanelRow>
+						<PanelRow>
+							<ToggleControl
+								label="canEditImage"
+								help="Can the user change the image?"
+								checked={ attributes.canEditImage }
+								onChange={ ( canEditImage ) =>
+									setAttributes( { canEditImage } )
+								}
+							/>
+						</PanelRow>
+						<PanelRow>
+							<SelectControl
+								label="tag"
+								help="How should the media be output?"
+								value={ attributes.tag }
+								options={ [
+									{ label: 'figure', value: 'figure' },
+									{ label: 'img', value: 'img' },
+									{ label: 'picture', value: 'picture' },
+								] }
+								onChange={ ( tag ) => setAttributes( { tag } ) }
+							/>
+						</PanelRow>
+						<PanelRow>
+							<TextControl
+								label="size"
+								help="Use WordPress image size identifier: `thumbnail`, `medium`, `large`, `full`, `your-custom-size`..."
+								value={ attributes.size }
+								onChange={ ( size ) =>
+									setAttributes( { size } )
+								}
+							/>
+						</PanelRow>
+						<PanelRow>
+							<TextControl
+								label="altSource"
+								help="Where do we get the alternative text? Leave blank to allow users to enter their own text. To pull from the media object use `alt`, `caption`, `title`, or `description`."
+								value={ attributes.altSource }
+								onChange={ ( altSource ) =>
+									setAttributes( { altSource } )
+								}
+							/>
+						</PanelRow>
+						<PanelRow>
+							<TextControl
+								label="className"
+								help="Class(es) to add to the component. Default is undefined."
+								value={ attributes.className }
+								onChange={ ( className ) =>
+									setAttributes( { className } )
+								}
+							/>
+						</PanelRow>
+						<PanelRow>
+							<TextControl
+								label="mediaId"
+								help="@todo"
+								value={ attributes.mediaId }
```

Comment:
There is a `NumberControl` component that might handle this better. The issues is this component is meant to handle strings but we want mediaId to be an integer. So it causes a few glitchy issues and we have to compensate for that. 

```suggestion
								value={ attributes.mediaId || ' ' } // If meidaId is not set or undefined, show an empty string.
```